src/driver/main.cpp

```diff
@@ -96,7 +101,15 @@ struct loader
            ap.help("Dim of a parameter (format: \"@name d1 d2 dn\")"),
            ap.append(),
            ap.nargs(2));
-
+        ap(dyn_param_dims,
+           {"--dyn-input-dim"},
+           ap.help("Dynamic dimensions of a parameter (format: \"@name\" \"{min1, max1, [opt1_1, "
```

Comment:
Looks like `opts ` be passed as {opt1, opt2, ..} with curly braces instead of square brackets.  We should put that here.

Reply:
It can be either curly braces or square brackets.

Reply:
okay. I would find using  `{}` more convenient. is it possible to add that to `help` ?

Reply:
I would rather use json for this instead of custom parsing: `@name '[{max:1,min:64,optimals:[1,2,4,8]},3,224,224]'`. Then you can convert it from json with something like:

```cpp
auto v = from_json_string(convert_to_json(s));
std::vector<migraphx::shape::dynamic_dimension> dyn_dims;
std::transform(v.begin(), v.end(), std::back_inserter(dyn_dims), [&](auto x) {
    if (x.is_object()) 
        return from_value<migraphx::shape::dynamic_dimension>(x);
    auto d = x.to<std::size_t>();
    return migraphx::shape::dynamic_dimension{d, d};
});
```

Reply:
This also allows specifying the fixed dimensions with just one integer.